core/clustersmngr/clustersmngr.go

```diff
@@ -90,7 +90,9 @@ func ClientConfigWithUser(user *auth.UserPrincipal) ClusterClientConfigFunc {
 			}).DialContext,
 		}
 
-		if user.Token != "" {
+		if user.Token == "" && user.ID == "" {
```

Comment:
Maybe we could move this to `.Validate()` on the Principal?

Reply:
We could but as far as I'm aware nothing else uses this and it feels like it makes more sense to keep it inline where it's clear what's being checked for and why.